devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java

```diff
@@ -91,7 +132,32 @@ public boolean addExtensions(final Set<String> extensions) throws IOException {
             writer.write(pom, pomOutputStream.toString("UTF-8"));
         }
 
-        return updated;
+        return new AddExtensionResult(updated, success);
+    }
+
+    private boolean addDependency(List<Dependency> dependenciesFromBom, Extension extension) {
+        if (!MojoUtils.hasDependency(model, extension.getGroupId(), extension.getArtifactId())) {
+            print(OK + " Adding extension " + extension.managementKey());
+            model.addDependency(extension
+                    .toDependency(containsBOM(model) &&
+                            isDefinedInBom(dependenciesFromBom, extension)));
+            return true;
+        } else {
+            // Don't set success to false as it's should be idempotent.
```

Comment:
you say don't set it to false but you do set it to false. I'm missing something.

Reply:
wrong place for this comment.

Reply:
Fixed.